src/vaadin-radio-button-mixin.ts

```diff
@@ -0,0 +1,141 @@
+import { LitElement, html, property, PropertyValues } from 'lit-element';
+import { Constructor } from '@vaadin/mixin-utils';
+import { ActiveStateMixin } from '@vaadin/active-state-mixin/active-state-mixin.js';
+import { ActiveStateClass } from '@vaadin/active-state-mixin/active-state-class.js';
+import { CheckedStateMixin, CheckedStateInterface } from '@vaadin/checked-state-mixin/checked-state-mixin.js';
+import { ControlStateMixin, ControlStateInterface } from '@vaadin/control-state-mixin/control-state-mixin.js';
+import { DisabledStateMixin, DisabledStateInterface } from '@vaadin/disabled-state-mixin/disabled-state-mixin.js';
+import { FocusVisibleMixin, FocusVisibleInterface } from '@vaadin/focus-visible-mixin/focus-visible-mixin.js';
+import { KeyboardClass } from '@vaadin/keyboard-mixin/keyboard-class.js';
+import { SlottedLabelMixin } from '@vaadin/slotted-label-mixin/slotted-label-mixin.js';
+import { RadioButtonMixinClass } from './vaadin-radio-button-mixin-class';
+
+export interface RadioButtonInterface {
+  value: string | null | undefined;
+}
+
+export type RadioButton = RadioButtonInterface &
+  DisabledStateInterface &
+  FocusVisibleInterface &
+  ControlStateInterface &
+  CheckedStateInterface &
+  ActiveStateClass &
+  KeyboardClass &
+  RadioButtonMixinClass;
+
+export const RadioButtonMixin = <T extends Constructor<LitElement>>(base: T): T & Constructor<RadioButton> => {
+  class RadioButton extends SlottedLabelMixin(
+    CheckedStateMixin(ControlStateMixin(FocusVisibleMixin(ActiveStateMixin(DisabledStateMixin(base)))))
+  ) {
+    /**
+     * Used for mixin detection because `instanceof` does not work with mixins.
+     */
+    static hasRadioButtonMixin = true;
```

Comment:
Yes it's for a different purpose. `wasApplied` is only used for deduping mixins, like `KeyboardMixin`, that must be only applied once per prototype chain.

We have this static field also in `SelectedStateMixin`.

Whether all the mixins should be deduping is a different topic. Maybe they should.

Reply:
Done by adding a KeyboardMixin.

Reply:
Is there a drawback for having all mixins being deduping (or just applied with the helper)? If we did that we would not need these separate `has*Mixin` properties as we could just use the `wasApplied()` method to check the same thing without having a unique static property in all mixins that we might need to check.

I think it's a bit weird that now we have 2 different ways to check if a mixin has been applied. Even if mixin-utils was originally made with deduping in mind, it would seem to work for other more general purpose use cases too like this case of `filterRadioButtons` method in radio group.

Reply:
Let's continue discussing this in https://github.com/vaadin/component-mixins/issues/60